raster-foundry/raster-foundry
# Create tool-run listing and search page

## Overview

This PR adds a tool-run listing page. It also...

Begins the process of renaming 'tools' to 'templates' and 'tool-runs' to 'tools'. I think there are still some questions as to this being the right verbiage (does 'analysis' make more sense than 'tool').

Gets rid of the market route and moves both 'template' and 'tool' listing pages to the `lab` route. There are also some other route changes to make this possible. The current routing of the lab is now:

- `lab/browse/templates` to view templates
- `lab/browse/tools` to view tools
- `lab/build/<templateid>/run/<toolid>`

This last route above will probably not stick around due to an impending redesign.

This PR also fixes a call to a no longer existing method on the project service that caused source nodes to not properly fetch their project. This one was weird as it seemed to work intermittently when I think it should have not worked at all. 

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

![image](https://user-images.githubusercontent.com/2442245/31406633-1ace21a2-add0-11e7-920f-d176e4323d84.png)

![image](https://user-images.githubusercontent.com/2442245/31406648-2677f442-add0-11e7-8cda-2e9452ddf811.png)

### Notes

**The searching is not yet functional**. In order for the search to work, we need to at least have fields on the tool-run entity to search by. We don't currently have them. See #2628. Due to this, we list the tool-run ID as the tool-run name.

Additionally, our front-end search strategy is not really viable for searching tool-runs, which will both be numerous and large, considering that they now contain the full AST. As such, we need to implement back-end searching on the `tool-runs` endpoint.

Older tool-runs, while visible in the UI, may not load properly inside the lab if they were created before some tool-run refactoring.

## Testing Instructions

 * Navigate around the app and ensure that links to the lab are functional with the new routing
 * Make sure the tool (template) searching still works
 * Try the tool-run (tool) listing and try loading up some tool-runs

Closes #2607
